### PR TITLE
API routes for tests

### PR DESCRIPTION
**Pull request by super9user**
- Added model & basic routes for the tests-buckets service.
- Original `Tests` route is renamed to `TestGroups`

### CODE REVIEW

File: lib/routes/test.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	TestModel.find({uuid: req.params.uuid}, function(err, tests) {
+		if (err) {
+			res.statusCode = 500;
+			return res.json({
+				error: err
+			});
+		}
+		if(tests.length !== 0) {
+			res.statusCode = 200;
+			res.json(tests[0]);
```
Comment (mpahuja) [inline lib/routes/test.js:13]: do we always want to return the first one? can this be any test in the list?

File: lib/routes/test.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+						error: err
+					});
+				}
+				res.statusCode = 200;
+				res.json(t);
+			})
+		} else {
+			res.statusCode = 404;
+			return res.json({
+				error: "Resource not found."
```
Comment (mpahuja) [inline lib/routes/test.js]: can we also return req.body.name?

File: lib/routes/test.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+			test.remove(function (err) {
+				if (err) {
+					res.statusCode = 500;
+					return res.json({
+						error: err
+					});
+				}
+				res.statusCode = 200;
+				return res.json({
+					message: "User successfully deleted."
```
Comment (mpahuja) [inline lib/routes/test.js]: Test successfully deleted or req.params.uuid deleted successfully?

File: lib/routes/test.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	return true;
+}
+
+
+function register(app) {
+	app.get('/tests/', getAll);
+	app.get('/tests/:uuid', getOne);
+	app.post('/tests/', createOne);
+	app.put('/tests/:uuid', updateOne);
+	app.delete('/tests/:uuid', deleteOne);
```
Comment (mpahuja) [inline lib/routes/test.js:154]: we can also have /POST, createAll, which takes a json object and creates multiple test records at once

File: lib/routes/test.js (hunk 1 above)
Comment (super9user) [inline lib/routes/test.js:13]: Uuid's are unique. So it will be always the first one.

File: lib/routes/test.js (hunk 4 above)
Comment (super9user) [inline lib/routes/test.js:154]: It would be hard to report back errors when multiple records are passed.

**Pull request closed without merging**